presto-hive/src/main/java/com/facebook/presto/hive/ColumnarBinaryHiveRecordCursor.java

```diff
@@ -184,64 +188,65 @@ public ColumnarBinaryHiveRecordCursor(RecordReader<K, BytesRefArrayWritable> rec
             types[i] = typeManager.getType(column.getTypeSignature());
             hiveTypes[i] = column.getHiveType();
 
-            if (!column.isPartitionKey()) {
+            if (!column.isPartitionKey() && !column.isHidden()) {
```

Comment:
I would move these 3 lines below `isPrefilledColumn[i] = ...` and use `isPrefilledColumn[i]` as the condition.